src/test/crossFetchNode.test.ts

```diff
@@ -274,3 +274,110 @@ test('should not fail when arbitrary parameters are configured in fetchOptions',
   });
   expect(response).toEqual({});
 });
+
+test('throws on responses other than and 304 errors', async () => {
+  nock('https://localhost:3000')
+    .get(
+      `/search/shopper-search/v1/organizations/${config.parameters.organizationId}/product-search?siteId=${config.parameters.siteId}&q=sony`
+    )
+    .matchHeader('authorization', 'Bearer test-auth')
+    .reply(
+      400,
+      {content: 'not empty'},
+      {'content-type': 'application-json charset=UTF-8'}
+    );
+
+  const standardConfig: ClientConfigInit<TestConfigParameters> = {...config};
+  const client = new ShopperSearch(standardConfig);
+  const response = async () =>
+    client.productSearch({
+      parameters: {q: 'sony'},
+      headers: {authorization: 'Bearer test-auth'},
+    });
+
+  expect(response).rejects.toThrow();
```

Comment:
```suggestion
  await expect(response).rejects.toEqual(something);
```
1. `expect(promise).rejects` and `expect(promise).resolves` return a Promise, so you must `await` it, otherwise the checks are never actually done.
2. `expect(promise).rejects` basically unwraps the rejected error so that you can make assertions about it. You should make some assertions about it.